src/db.ts

```diff
@@ -1,27 +1,204 @@
+        Questions?: Array<string>,
+    }): Promise<void> {
+        const quizModel = getModelForClass(QuizSchema);
+        try {
+            const res = await quizModel.updateOne({ _id: quizId }, updateData);
+            console.log(`Quiz updated with id (handler): ${quizId}`);
+        } catch (error) {
+            console.error("Quiz update error (handler):", error);
+        }
+    }
+
+    public async createQuestionHandler(questionData: QuestionSchema): Promise<void> {
+        try {
+            const questionModel = getModelForClass(QuestionSchema);
+            const question = new questionModel({ Question: questionData.Question, PossibleAnswers: questionData.PossibleAnswers, CorrectAnswer: questionData.CorrectAnswer, QuestionType: questionData.QuestionType});
+            await question.save();
+            console.log(`New question created with id (handler): ${question._id}`);
+        } catch (error) {
+            console.error("Question create error (handler):", error);
+        }
+    }
+
+    public async updateQuestionHandler(questionId: string, updateData: {
+        Question?: string,
+        PossibleAnswers?: Map<string, string>,
+        CorrectAnswer?: string,
+        QuestionType?: string,
+    }): Promise<void> {
+        const questionModel = getModelForClass(QuestionSchema);
+        try {
+            const res = await questionModel.updateOne({ _id: questionId }, updateData);
+            console.log(`Question updated with id (handler): ${questionId}`);
+        } catch (error) {
+            console.error("Question update error (handler):", error);
+        }
+    }
+
 }
 
-export const dbConnect = async () => {
-    await run().catch(console.dir);
+class dbInterface {
+    private db: dbHandler = dbHandler.getInstance();
+
+    public async createRoomRecord(hostWsId: string, playerIds: Array<string>, roomState: string): Promise<void> {
+        const roomRecord: RoomSchema = {
+            HostWsId: hostWsId,
+            PlayerIds: playerIds,
+            RoomState: roomState,
+        };
+
+        try {
+            await this.db.createRoom(roomRecord);
+            console.log(`Room created (interface)`);
+        } catch (error) {
+            console.error("Failed to create room error (interface):", error);
+        }
+    }
+
+    public async updateRoomState(roomId: string, newState: string) {
+        const updateObject = { RoomState: newState };
+        await this.db.updateRoomRecord(roomId, updateObject);
+    }
+
+    public async updateRoomHostId(roomId: string, newHost: string) {
+        const updateObject = { HostWsId: newHost };
+        await this.db.updateRoomRecord(roomId, updateObject);
+    }
+
+    public async updateRoomPlayerIds(roomId: string, playerIds: Array<string>) {
+        const updateObject = { PlayerIds: playerIds };
+        await this.db.updateRoomRecord(roomId, updateObject);
+    }
+
+    public async createQuizList(questions: Array<string>): Promise<void> {
```

Comment:
[NIT] this should be `createQuiz`

Reply:
is `questions` an array of ids of questions in the database?
e.g: `['1234', '2345', ...]`

Reply:
yes, questions in this context are a list of questions indexes/ids.